File: 백준/Silver/1427. 소트인사이드/소트인사이드.java
import java.io.*;
import java.util.*;

public class Main{
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String str = br.readLine();
        int[] numbers = new int [str.length()];
        for(int i=0; i<str.length(); i++){
            numbers[i] = Integer.parseInt(str.substring(i,i+1));
        }
        for(int i=0; i<str.length(); i++){
            int max = i;
            for(int j= i+1; j<str.length(); j++){
                if(numbers[max]<numbers[j]){
                    max = j;
                }
            }
            if(numbers[max]>numbers[i]){
                int temp = numbers[i];
                numbers[i] = numbers[max];
                numbers[max] = temp;
            }
        }
        for(int i=0; i<str.length(); i++){
            System.out.print(numbers[i]);
        }
    }
}

File: 백준/Silver/1427. 소트인사이드/소트인사이드.kt
import java.io.*
import java.util.*

fun main(){
    val bf = BufferedReader(InputStreamReader(System.`in`))
    val N = bf.readLine()
    val numbers = N.map{ it.toString().toInt() }.sortedDescending()
    //문자열에 Map을 취할 경우 각 문자에 대해 람다표현식 적용
    //toString으로 Char타입을 String으로 변환해준다.
    val results = numbers.joinToString("")     
    println(results)     
}

  
